fictadvisor-web/src/components/pages/admin/admin-main/AdminHomePage.tsx

```diff
@@ -0,0 +1,339 @@
+'use client';
+import React, { FC, useEffect, useState } from 'react';
+import { Box, CardHeader, Stack, Typography } from '@mui/material';
+import { Field, Form, Formik } from 'formik';
+import { useRouter } from 'next/navigation';
+
+import Button from '@/components/common/ui/button-mui';
+import {
+  ButtonColor,
+  ButtonSize,
+} from '@/components/common/ui/button-mui/types';
+import Switch from '@/components/common/ui/form/switch';
+import { SwitchLabelPlacement } from '@/components/common/ui/form/switch/types';
+import Input from '@/components/common/ui/form/with-formik/input/FormikInput';
+import FormikTextArea from '@/components/common/ui/form/with-formik/text-area';
+import { useToastError } from '@/hooks/use-toast-error/useToastError';
+import PageTextsAPI from '@/lib/api/page-texts/PageTextsAPI';
+import {
+  GetPageTextsResponse,
+  PageText,
+} from '@/lib/api/page-texts/types/GetPageTextsResponse';
+import StudentResourcesAPI from '@/lib/api/student-resources/StudentResourcesAPI';
+import {
+  GetStudentResourcesResponse,
+  StudentResource,
+} from '@/lib/api/student-resources/types/GetStudentResourcesResponse';
+
+import * as styles from './AdminHomePage.styles';
+interface FormValues {
+  [key: string]: unknown;
+  switch_primary: boolean;
+  switch_secondary: boolean;
+}
+
+interface TextsProperties {
+  key: string;
+  value: string;
+  link: string | null;
+  isShown: boolean;
+}
+
+interface ChangedPageTexts {
+  mainpage_title?: { value: string; link: string; isShown: boolean };
+  mainpage_description?: { value: string; link: string; isShown: boolean };
+  mainpage_primary?: { value: string; link: string; isShown: boolean };
+  mainpage_secondary?: { value: string; link: string; isShown: boolean };
+  mainpage_studentresources_title?: {
+    value: string;
+    link: string;
+    isShown: boolean;
```

Comment:
you can move this type: { value: string; link: string; isShown: boolean } to additional interface to avoid duplicating code